cmd/thanos/bucket.go

```diff
@@ -375,6 +379,38 @@ func registerBucketWeb(m map[string]setupFunc, root *kingpin.CmdClause, name str
 	}
 }
 
+func registerBucketReplicate(m map[string]setupFunc, root *kingpin.CmdClause, name string, objStoreConfig *extflag.PathOrContent) {
+	cmd := root.Command("replicate", "Replicate data from one object storage to another")
+	httpMetricsBindAddr, _ := regHTTPFlags(cmd)
+	toObjStoreConfig := regCommonObjStoreFlags(cmd, "-to", false, "The object storage which replicate data to.")
+	resolution := cmd.Flag("resolution", "Only blocks with this resolution will be replicated.").Default(strconv.FormatInt(downsample.ResLevel0, 10)).Int64()
+	compaction := cmd.Flag("compaction", "Only blocks with this compaction level will be replicated.").Default("1").Int()
+	matcherStrs := cmd.Flag("matcher", "Only blocks whose labels match this matcher will be replicated.").PlaceHolder("key=\"value\"").Strings()
+	singleRun := cmd.Flag("single-run", "Run replication only one time, then exit.").Default("false").Bool()
+
+	m[name+" replicate"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, tracer opentracing.Tracer, _ bool) error {
+		matchers, err := replicater.ParseFlagMatchers(*matcherStrs)
+		if err != nil {
+			return errors.Wrap(err, "parse block label matchers")
+		}
+
+		return replicater.RunReplicate(
```

Comment:
```suggestion
		return replicator.RunReplicate(
```

Reply:
I use "replicater" as package name, i google it and "replcater" and "replicator" are both right grammar, how to choose?

Reply:
It seems like Oxford's dictionary has no entry for `replicater`: https://www.oxfordlearnersdictionaries.com/spellcheck/english/?q=replicater
Only `replicator`: https://www.oxfordlearnersdictionaries.com/spellcheck/english/?q=replicator

Also, I'm not a native English speaker but `replicator` seems the correct version here to me.

Reply:
This web helps me! Thanks a lot!